### PR TITLE
Fix/work package index url params

### PR DESCRIPTION
**Pull request by manwithtwowatches**
https://www.openproject.org/work_packages/14792
https://www.openproject.org/work_packages/14451

What I'm trying to achieve here is to use ui-router to initialise WorkPackagesListController whenever the URL params change. Whenever the query changes the URL gets updated with the query state. This is represented in two params - one for columns and displaySums (which is ignored by ui-router) and one for all the rest. The reason for this is that changing columns or displaySums shouldn't trigger the fetching of work packages.

If a query_id is given in the params then that query will be loaded but it will also be augmented with any specific query params if they are also present.

The main problem that I see with this approach is that when the controller is being re-initialised the filters disappear from from the view and then become visible again once the query has been loaded. To work around this would probably involve a pretty major refactor which I wouldn't really know how to do off the top of my head.

### CODE REVIEW

File: karma/tests/helpers/url-params-helper-test.js [hunk 1]
```diff
@@ -55,4 +55,98 @@ describe('UrlParamsHelper', function() {
...
+      var filter = {
+        modelName: 'sosse',
+        name: 'sosse_id',
+        type: 'list model',
+        operator: '=',
+        values: ['knoblauch']
+      };
+      query = new Query({
+        id: 1,
+        name: 'knoblauch sosse',
```
Comment (hschink) [inline karma/tests/helpers/url-params-helper-test.js:72]: There is a typo: `sosse` is actually written `soße` :smirk:

**Comment by hschink** [comment]
I just discussed the implementation with Richard and like to add a point here: Why do we need to use the ui rooter for updating the query at all? The only reason we need the query parameters in the URL are for _initializing_ the query on page load. But now we also use the query parameters for _updating_ the current query. I think _updating_ is a different purpose (in respect to _initializing_) and ui rooter is not the right approach to it. I would rather consider the previous means for updating the query the right approach. Opinions? :smirk:

**Comment by manwithtwowatches** [comment]
I generally agree. I'm going to make an alternative implementation on a different branch just now.

**Comment by mfrister** [comment]
Just a formal note: There are three WIP commits in here. If you don't think a commit on it's own is worth having it, please squash it together with other's into a minimal useful commit.

**Pull request closed without merging**